### ISSUE TITLE
EE 1.13.1 to 2.0.2 Data Migration "hung" at "Customer Attributes Step"

### ISSUE THREAD

**Issue by rickoconnor** [magento/data-migration-tool#75]
I have successfully run the settings migration and after resolving a few issues with the data migration I got to the  "Customer Attributes Step" and it is at 0% and has been for at least 30 minutes.  Is this normal or is there an issue?  There isn't an error message displayed.

**Comment by kennylawrence** [comment]
are you on a local LAMP environment or running it elsewhere? How big is your database? Mine was about 1.4GByte and some spots might take a few minutes, but it wasn't that bad. I couldn't get it to work on Windows, but it worked on the Linux host. I DID think it hung at first but a few minutes went by and I went back to the terminal and noticed that the progress meter was indeed moving.

**Comment by rickoconnor** [comment]
It's running on an AWS instance running Centos 6.7. The original, 1.13, DB is 9.3 GB.

**Comment by kennylawrence** [comment]
Hmm that might be normal then.. mine was at 0 for a few minutes. The progress bar isn't very linear it seems. On Windows I was getting the mysql server disconnecting. I think you would have seen an error by now if there was a problem. Let us know if it does something within an hour maybe. There is a verbose mode but unfortunately it only affects the error report after an error has occurred. I guess another option is to go edit the php files and have it spit out a number every time it copies something new, or every 100 things

**Comment by rickoconnor** [comment]
Thank you very your responses and suggestions.  I will let you know if anything happens.

**Comment by rickoconnor** [comment]
So far:
[2016-04-05 17:17:06][INFO][mode: data][stage: data migration][step: Customer Attributes Step]: started
0% [>---------------------------] Remaining Time:

**Comment by rickoconnor** [comment]
The migration is still "hung" at the "Customer Attributes Step", but it looks like the four tables it copies data from->to are fillled
enterprise_customer_sales_flat_order -> magento_customercustomattributes_sales_flat_order: 217,466 -> 221,599
enterprise_customer_sales_flat_order_address -> magento_customercustomattributes_sales_flat_order_address: 434,025 -> 434,025
enterprise_customer_sales_flat_quote -> magento_customercustomattributes_sales_flat_quote 12,954 -> 12,954
enterprise_customer_sales_flat_quote_address -> magento_customercustomattributes_sales_flat_quote_address: 19,743 -> 19743
I got the table names from /mnt/www/roconnor/current/vendor/magento/data-migration-tool/etc/ee-to-ee/customer-attr-map.xml.dist

**Comment by rickoconnor** [comment]
Left it running overnight and it failed with:
[PDOException]
  SQLSTATE[08S01]: Communication link failure: 1153 Got a packet bigger than 'max_allowed_packet' bytes

**Comment by Caprico85** [comment]
I had the same problem. I was able to fix it (or work around it) by setting
```
<bulk_size>100</bulk_size>
<direct_document_copy>1</direct_document_copy>
```
in my config.xml.

**Comment by rickoconnor** [comment]
I made the changes and am trying the data migration again.  Thanks.

**Comment by rickoconnor** [comment]
Ended up getting the same error.  Trying to increase max_allowed_packet.

**Comment by rickoconnor** [comment]
Increased the max_allowed_packet, but that did not work.

**Comment by ilol** [comment]
MAGETWO-52582

**Comment by kennpie** [comment]
same issue - hanging at Customer Attributes Step, migrating from v 1.7.0.2 to v 2.0.7, AWS Ubuntu instance. For me it seems to hang right as it's finishing that step. It's been hung for over an hour now.

[2016-06-13 17:14:15][INFO][mode: data][stage: data migration][step: EAV Step]: started
100% [============================] Remaining Time: 1 sec
[2016-06-13 17:14:17][INFO][mode: data][stage: volume check][step: EAV Step]: started
100% [============================] Remaining Time: 1 sec
[2016-06-13 17:14:17][INFO][mode: data][stage: data migration][step: Customer Attributes Step]: started
[2016-06-13 17:14:18][DEBUG][mode: data][stage: data migration][step: Customer Attributes Step][table: customer_entity]: migrating
100% [============================] Remaining Time: 1 sec

**Comment by kennpie** [comment]
Yesterday I let the above hang persist for as long as it wanted; after about an hour or so it errored out with

proc_open(): fork failed - Cannot allocate memory
#0 [internal function]: Composer\Util\ErrorHandler::handle(2, 'proc_open(): fo...', '/var/www/html/v...', 974, Array)
#1 /var/www/html/vendor/symfony/console/Symfony/Component/Console/Application.php(974): proc_open('stty -a | grep ...', Array, NULL, NULL, NULL, Array)
#2 /var/www/html/vendor/symfony/console/Symfony/Component/Console/Application.php(784): Symfony\Component\Console\Application->getSttyColumns()
#3 /var/www/html/vendor/symfony/console/Symfony/Component/Console/Application.php(745): Symfony\Component\Console\Application->getTerminalDimensions()
#4 /var/www/html/vendor/symfony/console/Symfony/Component/Console/Application.php(675): Symfony\Component\Console\Application->getTerminalWidth()
#5 /var/www/html/vendor/symfony/console/Symfony/Component/Console/Application.php(133): Symfony\Component\Console\Application->renderException(Object(PDOException), Object(Symfony\Component\Console\Output\StreamOutput))
#6 /var/www/html/bin/magento(25): Symfony\Component\Console\Application->run()

I increased the memory of the AWS instance, and allocated more memory to MySQL, and it seemed to get past that point finally... however today, trying to repeat after a reset of the Magento2 db... it's hanging at the same point. I'm going to try to let it error out again, but that's an hour+ down the tubes... can someone offer diagnostic help, or suggest a way to have the thing detect the problem and quit sooner, with a better indication of where the problem is?  Yes I'm using -vvv. Thanks.

**Comment by ilol** [comment]
Thank you, @kennpie. Believe you information will help with this issue fixing.

**Comment by victor-v-rad** [comment]
@rickoconnor 
The issue was fixed in the latest release of the tool 2.1.1
Please recheck

**Comment by victor-v-rad** [comment]
I am closing the issue. Feel free to reopen if it is still actual
